pyilc/input.py

```diff
@@ -163,6 +167,31 @@ def __init__(self, input_file=None):
         self.N_freqs = p['N_freqs']
         assert type(self.N_freqs) is int and self.N_freqs > 0, "N_freqs"
 
+        # wavelet_beam_criterion, set to 1e-3 by default. This removes frequencies from the NILC
+        # whose beams are a certain fraction smaller than the appropriate needlet filter within the range
+        # of ells appropriate for the filter.
+        if 'wavelet_beam_criterion' in p.keys():
+            self.wavelet_beam_criterion = p['wavelet_beam_criterion']
+        else:
+            self.wavelet_beam_criterion = 1.e-3
+
+        # override_N_freqs_to_use OVERRIDES information from wavlet_beam_criterion
+        # and allows you to explicitly specify how many frequencies to use at each wavelet scale
+        # this should be a list of ints, of length N_scales, where each entry in the list specifies 
+        # how many frequency channels one should use at the scale corresponding to that entry. 
+        # if the entry is less than N_freqs, the lowest resolution maps will be dropped from the NILC
+        # such that there are the appropriate number of frequency channels used in each scale
```

Comment:
OK, so the counting is such that we count starting from the highest-resolution channel and then go in order of decreasing resolution... perhaps clarify this in the comments of pyilc_input_example_general.yml

Reply:
done